Src/main.c

```diff
@@ -132,17 +133,12 @@ int main(void) {
         float oncycles = adc_val / 1023; // basic conversion to decimal
```

Comment:
Maybe make the max value of the ADC a const?

Reply:
I would recommend looking at truncation, as may make your oncycles value incorrect over here. Looking at [this link](https://www.quora.com/In-C-integer-division-truncates-Please-explain-this-line#:~:text=When%20integers%20are%20divided%20in,is%20ceiling%20of%20the%20output%20) should help you understand truncation better.

To solve this, you can look into different datatypes and see what happens to decimal places when you divide them. [This link should get you started](https://stackoverflow.com/questions/60088323/how-i-can-prevent-integers-from-truncating-value-while-division-in-c)